src/filesystem.cpp

```diff
@@ -1002,18 +1004,34 @@ std::string get_exe_dir()
 	bfs::path exe(process_path);
 	return exe.parent_path().string();
 #else
+	// first check /proc
 	if(bfs::exists("/proc/")) {
 		bfs::path self_exe("/proc/self/exe");
 		error_code ec;
 		bfs::path exe = bfs::read_symlink(self_exe, ec);
-		if(ec) {
-			return std::string();
+		if(!ec) {
+			return exe.parent_path().string();
 		}
+	}
 
-		return exe.parent_path().string();
-	} else {
-		return get_cwd();
+	// check the PATH for wesnoth's location
+	// versionless
+	std::string exe = filesystem::base_name(filesystem::get_program_invocation("wesnoth"));
```

Comment:
It's never a path now, right? So no point in calling base_name.

Reply:
Updated.